### DISCUSSION TITLE
File inputs are empty when returning to the previous page in a multi-page form

### DISCUSSION THREAD

**Discussion by FrDH** [solspace/craft-freeform#450] [Questions & Answers] [answered]
**Describe the bug or issue you're experiencing**
In a multi-page form, file-type inputs appear to be empty after returning back from the next page.

**Steps to reproduce**

1. Create a multi-page form
2. Create a file input and place it on the first page
3. Go to the form on the frontend
4. Select a file in the file input and submit to the next page
5. Go back to the first page, the file input is empty

**Expected behavior**
A textual representation of the uploaded file, with a button to remove it.

**Craft & Plugin Info (please complete the following information):**

- Craft Version: 3.7.44
- Freeform Version: 3:13:15
- Freeform Edition: PRO
- Fresh Install or Upgrade: Fresh install

**Additional context**
We render our fields like this:
```
field.renderInput({
     class: inputClass,
     inputAttributes: inputAttrs,
     useRequiredAttribute: true
})
```

**Comment by kjmartens** [comment]
Hi @FrDH,

This is a browser limitation as the basic file upload field is too simple to allow this to work. However, this is one reason why we recently added the [Drag & Drop File Upload](https://docs.solspace.com/craft/freeform/v3/overview/fields.html#file-upload-drag-drop) field type. It's not only more user-friendly, but it also is advanced enough to keep track of uploaded files (even if temporarily before the form is fully submitted) and display them visually, etc. 🙂 

![field_file-upload-drag-drop 54471501](https://user-images.githubusercontent.com/1821674/189364927-44d5b4fe-54ae-48b1-a81f-a1888d285e7a.png)

**Reply by FrDH** [reply]
Thanks, I did not know the drag & drop field had that functionality.

**Comment by FrDH** [comment]
I finally got the time to give this a try. But the drag & drop field also does not show the uploaded images when I return to the page, I made a video to explain the issue.

https://user-images.githubusercontent.com/1387163/189843210-5a6c55b3-572c-41fc-be9d-81e1c17377c9.mov

**Comment by kjmartens** [comment]
Sorry for the delay @FrDH...

Are you able to check the browser console and share what the payload and network response are when going back in the form? 🙂

**Comment by FrDH** [comment]
A little late with my reply... :| But hope you're still willing to have a look at this.
Here's what I'm seeing in the console network tab (I removed the hashes):
```
{
  freeform-payload: (some giant hash)
  formHash: (a smaller hash)
  action: freeform/submit
  CRAFT_CSRF_TOKEN:  (another hash)
  freeform-action: back
  firstname: My first name
  lastname:  My last name
  (all other fields, just not the file-upload field)
} 
```

**Comment by FrDH** [comment]
Any updates on this?
Thank you.

**Comment by kjmartens** [comment]
Sorry for the delay @FrDH.

The File Upload Drag & Drop field asks for the uploaded files via another endpoint, so the form AJAX response does not contain anything for it. We're wondering if there's a problem with your Freeform scripts loading.

Can you install the demo templates (and use the Bootstrap test area) and see if the issue happens there as well? If it works correctly there, it will help us narrow things down to potentially being an issue in your template. 🙂

**Comment by FrDH** [comment, marked as answer]
I don't know what to say, the drag & drop field now works as expected. Maybe some update since this issue was opened fixed it.
One thing to note is that the upload only works if an asset source is selected for the form, however that setting is not required. (to be clear, we had one selected in this case 😃 )

Thank you for the help.

**Reply by kjmartens** [reply]
Thanks for letting me know. We'll check into it 🙂
